test/test_nested_dim.py

```diff
@@ -83,8 +82,7 @@ def test_mutate_array():
     arrs = Arrs(parent=root)
 
     arr = np.ones(arrs.arr.shape)
-    with pytest.raises(TypeError, match=r".*cannot be assigned to a DataTree.*"):
```

Comment:
I forget if I thought at first that this was an xarray limitation (i.e. dataset immutability) or just put this in to make the tests pass